plugins/TKLiveSync/TKLiveSync/TKLiveSync.m

```diff
@@ -16,67 +19,172 @@ static void tryExtractLiveSyncArchive()
+"Sync completed successfully.</br><a href=\"%@://reload\">Opening shortly in NativeScript."
+"</p><script>var xhttp = new XMLHttpRequest();xhttp.open('RESTART', '/', true);xhttp.send();</script></body></html>\n";
+
+static void restartApp()
+{
+    NSURL *restartURL = availableRestartURL();
+    if (restartURL) {
+        [UIApplication.sharedApplication openURL:restartURL];
+        exit(0);
+    }
+    
+    BOOL useSafariForRestart = [NSBundle.mainBundle.infoDictionary[@"SafariRestart"] boolValue];
+    if(!useSafariForRestart){
+        UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Application Restart Required"
+                                                        message:@"To view the current changes please restart the application"
+                                                       delegate:nil
+                                              cancelButtonTitle:@"OK"
+                                              otherButtonTitles:nil];
+        [alert show];
+        return;
     }
+    
+    // Starting a local web server used for a restart.
+    NSArray *bundleURLTypes = NSBundle.mainBundle.infoDictionary[@"CFBundleURLTypes"];
+    NSString *appScheme = [bundleURLTypes.firstObject[@"CFBundleURLSchemes"] firstObject];
+    NSString *html = [NSString stringWithFormat: kHTMLResponse, appScheme, appScheme];
+    [GCDWebServer setLogLevel:1];
+    if (!webServer) {
+        webServer = [[GCDWebServer alloc] init];
+        [webServer addDefaultHandlerForMethod:@"GET"
+                                 requestClass:[GCDWebServerRequest class]
+                                 processBlock:^GCDWebServerResponse *(GCDWebServerRequest* request) {
+                                     return [GCDWebServerDataResponse responseWithHTML:html];
+                                 }];
+        
+        [webServer addDefaultHandlerForMethod:@"RESTART"
+                                 requestClass:[GCDWebServerRequest class]
+                                 processBlock:^GCDWebServerResponse *(GCDWebServerRequest* request) {
+                                     exit(0);
+                                 }];
+    }
+    
+    NSError *error;
+    NSNumber *port = @(8081);
+    NSString *url = [NSString stringWithFormat:@"http://127.0.0.1:%@/1/", port];
+    BOOL isServerSuccessfullyStarted = webServer.isRunning;
+    
+    if (!isServerSuccessfullyStarted) {
+        isServerSuccessfullyStarted = [webServer startWithOptions:@{
+                                                                    GCDWebServerOption_AutomaticallySuspendInBackground: @(NO),
+                                                                    GCDWebServerOption_Port: port
+                                                                    }
+                                                            error:&error];
+    }
+    
+    if (!isServerSuccessfullyStarted || error) {
```

Comment:
As far as I can understand from [GCDWebServer source](https://github.com/swisspol/GCDWebServer/blob/master/GCDWebServer/Core/GCDWebServer.m#L761-L785) there is no such chance. Also, it seems they are relying on the returned boolean value to [determine whether a operation is successful](https://github.com/swisspol/GCDWebServer/blob/master/GCDWebServer/Core/GCDWebServer.m#L879) so we can remove the additional error check.